internal/exec/stages/disks/luks.go

```diff
@@ -531,3 +564,151 @@ func (d LuksDump) hasFlag(flag string) bool {
 	}
 	return false
 }
+
+// collect the cex card domains xx.xxxx
+func getAllCexOnlineAPQNs() (string, error) {
+	apDir := "/sys/bus/ap/devices/"
+	var ret string
+	// cards details are in form of link file points to card directory.
+	event, _ := os.ReadDir(apDir)
+	if len(event) == 0 {
+		return "", fmt.Errorf("adjunct processor not found")
+	}
+	// match the regexp for pattern xx.xxxx
+	for _, dir := range event {
```

Comment:
I'd just check upfront here if `dir.IsDir()` and continue if it's not.

Reply:
Done